crates/cxx-qt-gen/src/generator/cpp/property/signal.rs

```diff
@@ -3,27 +3,31 @@
 //
 // SPDX-License-Identifier: MIT OR Apache-2.0
 
-use syn::ForeignItemFn;
+use proc_macro2::Span;
+use syn::{ForeignItemFn, Ident};
 
 use crate::{
-    generator::naming::{property::QPropertyName, qobject::QObjectNames},
+    generator::naming::{property::QPropertyNames, qobject::QObjectNames, CombinedIdent},
     parser::signals::ParsedSignal,
 };
 
-pub fn generate(idents: &QPropertyName, qobject_idents: &QObjectNames) -> ParsedSignal {
+pub fn generate(idents: &QPropertyNames, qobject_idents: &QObjectNames) -> ParsedSignal {
     // We build our signal in the generation phase as we need to use the naming
     // structs to build the signal name
     let cpp_class_rust = &qobject_idents.name.rust_unqualified();
-    let notify_cpp = &idents.notify.cpp;
-    let notify_rust_str = idents.notify.rust.to_string();
+    let notify_cpp = &idents.notify.cxx_unqualified();
+    let notify_rust = idents.notify.rust_unqualified();
     let method: ForeignItemFn = syn::parse_quote! {
         #[doc = "Notify for the Q_PROPERTY"]
-        #[rust_name = #notify_rust_str]
-        fn #notify_cpp(self: Pin<&mut #cpp_class_rust>);
+        #[cxx_name = #notify_cpp]
+        fn #notify_rust(self: Pin<&mut #cpp_class_rust>);
```

Comment:
do we have a common way of using cxx_name in extern C++ blocks or cxx_name in extern Rust blocks and rust_name in the opposite ? eg which way round are we trying to write new code?

Reply:
With the new `Name` API, it makes more sense to do it this way around.

`rust_unqualified` returns an `Ident`, whilst `cxx_unqualified` returns a String.
This encourages the use of `cxx_name` with the `rust_name` as the main identifier.
As otherwise `parse_quote!` complains.

In most cases, we should just always emit both names, even if they're the same, it's easiest and most explicit.
With the new API the typing and naming all works out as long as you use the `cxx_name` attribute.

Reply:
cool, just wanted to check it's been thought about